front/src/applications/operationalStudies/utils.ts

```diff
@@ -209,3 +216,74 @@ export const convertDepartureTimeIntoSec = (departureTime: string) => {
 
 export const isInvalidName = (name?: string | null) =>
   !name || name.length > SMALL_INPUT_MAX_LENGTH;
+
+/**
+ * Check if the scheduled points are honored with a trainSchedule and a trainSummary
+ * @param trainResult
+ * @param trainSummary
+ * @returns true if the scheduled points are not honored
+ */
+export const isScheduledPointsNotHonored = (
+  trainSchedule: TrainScheduleResult,
+  trainSummary: Extract<SimulationSummaryResult, { status: 'success' }>
+): boolean => {
+  if (trainSummary.path_item_times_final.length !== trainSchedule.path.length) {
+    console.error();
+    throw new Error('Assertion failed');
+  }
+  if (!trainSchedule.schedule) return false;
+  return trainSchedule.schedule.some((schedule, index) => {
+    if (schedule.arrival === null || schedule.arrival === undefined) return false;
+    // need to add +1 because origin will never be in schedules
+    const arrivalTimeInMs = sToMs(ISO8601Duration2sec(schedule.arrival));
+    return (
+      Math.abs(arrivalTimeInMs - trainSummary.path_item_times_final[index + 1]) >=
+      ARRIVAL_TIME_ACCEPTABLE_ERROR_MS
+    );
+  });
+};
+
+export const getPathItemByIndexDict = (trainResult: TrainScheduleResult) =>
+  trainResult.path.reduce((acc, pathItem, index) => {
+    acc[pathItem.id] = index;
+    return acc;
+  }, {} as Dictionary<number>);
+
+/**
+ * Check if the train is too fast with a trainSchedule and a trainSummary
+ * @param trainResult
+ * @param trainSummary
+ * @returns true if the train is too fast
+ */
+export const isTooFast = (
+  trainSchedule: TrainScheduleResult,
+  trainSummary: Extract<SimulationSummaryResult, { status: 'success' }>
+): boolean => {
+  if (
+    trainSummary.path_item_times_final.length !== trainSummary.path_item_times_provisional.length
+  ) {
+    throw new Error('Assertion failed');
+  }
+
+  const toCheckPathItemIds = new Set(trainSchedule.margins?.boundaries);
+  trainSchedule.schedule?.forEach((schedule) => {
+    if (schedule.arrival !== null || schedule.stop_for !== null) {
```

Comment:
The above case handles `arrival === undefined` as well - is this something that can happen?

Reply:
I don't think so but with this condition I don't need to add the cast l237
[https://github.com/OpenRailAssociation/osrd/pull/8142/commits/464feb66f78a8ea158b4632afebe8a4c5799183d#r1705449975](url)

Reply:
I don't think so, but when in doubt I modified it